src/kibana/components/fancy_forms/kbn_form.js

```diff
@@ -20,7 +22,42 @@ define(function (require) {
       var count = self.errorCount();
       return count + ' Error' + (count === 1 ? '' : 's');
     };
+
+    self.$invalidModels = function () {
+      return _.reduce(self.$error, function (all, models) {
+        return all.concat(models ? models : []);
+      }, []);
+    };
+
+    self.$setTouched = function () {
+      self.$invalidModels().forEach(function (model) {
+        if (model.$setTouched) { // only kbnModels have $setTouched
+          model.$setTouched();
+        }
+
+        if (model.$invalidModels) {
```

Comment:
Seems a bit repetitive, with the other one, perhaps an or statement is more appropriate here?